libraries/chain/db_getter.cpp

```diff
@@ -164,7 +164,8 @@ std::set<son_id_type> database::get_sons_to_be_deregistered()
 
    for( auto& son : son_idx )
    {
-      if(son.status == son_status::in_maintenance)
+      if((son.status == son_status::in_maintenance) ||
+         (son.status == son_status::request_maintenance))
```

Comment:
Deregistration is 12 hours from the time the SON went to in_maintenance.
So this check may not be necessary.

Reply:
It is for the case when maintenance block is going after 12 hours after the start of the request_maintenance state (some error in network) and SON wasn't moved to in_maintenance state,  but 12 hours are passed.

Reply:
Actually it is expected that the SON continues to send heartbeats in request_maintenance state. So it is expected that the SON continues to work in full capacity till the next chain maintenance.

Reply:
Ah,  I see now,  I have missed this moment. I will fix it